packages/components/src/templates/next/layouts/Collection/utils/sortCollectionItems.ts

```diff
@@ -0,0 +1,96 @@
+import type { GetCollectionItemsProps } from "./getCollectionItems"
+import type { AllCardProps } from "~/interfaces"
+
+export type SortableCardProps = AllCardProps & {
+  rawDate?: Date
+}
+
+export interface SortCollectionItemsProps
+  extends Pick<GetCollectionItemsProps, "sortBy" | "sortDirection"> {
+  items: SortableCardProps[]
+}
+
+// Sort by last updated date, tiebreaker by title
+const sortCollectionItemsByDate = ({
+  items,
+  sortDirection = "desc",
+}: Omit<SortCollectionItemsProps, "sortBy">) => {
+  return items.sort((a, b) => {
+    const bothHaveDates = a.rawDate instanceof Date && b.rawDate instanceof Date
+    const bothSameDatetime = a.rawDate?.getTime() === b.rawDate?.getTime()
+
+    if (bothHaveDates && !bothSameDatetime) {
+      // Type assertion because TS control-flow narrowing only works when
+      // check is done inline and not when we define the variable
+      const aDate = a.rawDate as unknown as Date
+      const bDate = b.rawDate as unknown as Date
+
+      switch (sortDirection) {
+        case "asc":
+          return aDate.getTime() >= bDate.getTime() ? 1 : -1
+        case "desc":
+          return aDate.getTime() <= bDate.getTime() ? 1 : -1
+        default:
+          const exhaustiveCheck: never = sortDirection
+          return exhaustiveCheck
+      }
+    }
+
+    const bothNoDates = a.rawDate === undefined && b.rawDate === undefined
+    if ((bothHaveDates && bothSameDatetime) || bothNoDates) {
+      return a.title.localeCompare(b.title, undefined, { numeric: true })
+    }
+
+    return a.rawDate instanceof Date ? -1 : 1
+  }) as AllCardProps[]
+}
+
+// Sort by title, tiebreaker by last updated date
+const sortCollectionItemsByTitle = ({
+  items,
+  sortDirection = "asc",
+}: Omit<SortCollectionItemsProps, "sortBy">) => {
+  return items.sort((a, b) => {
+    const bothHaveDates = a.rawDate instanceof Date && b.rawDate instanceof Date
+    const bothNoDates = a.rawDate === undefined && b.rawDate === undefined
+
+    if ((bothHaveDates && a.title !== b.title) || bothNoDates) {
+      switch (sortDirection) {
+        case "asc":
+          return a.title.localeCompare(b.title, undefined, { numeric: true })
+        case "desc":
+          return b.title.localeCompare(a.title, undefined, { numeric: true })
+        default:
+          const exhaustiveCheck: never = sortDirection
+          return exhaustiveCheck
+      }
+    }
+
+    if (bothHaveDates && a.title === b.title) {
+      // Type assertion because TS control-flow narrowing only works when
+      // check is done inline and not when we define the variable
+      const aDate = a.rawDate as unknown as Date
+      const bDate = b.rawDate as unknown as Date
+
+      return aDate.getTime() < bDate.getTime() ? 1 : -1
```

Comment:
if same title, same date, shouldnt we return 0?

Reply:
didn't do that because was thinking it shouldn't affect the sorting order since it will still be determistic, but addressed it in to improve clarity https://github.com/opengovsg/isomer/pull/1212/commits/a7bada148ac0ce87a01edc7d51d1e7ec1a4f4f49